setup.py

```diff
@@ -1,6 +1,57 @@
+import io
+import re
 from setuptools import setup, find_packages
-import sys
 
+
+def read_file(filename, **kwargs):
+    encoding = kwargs.get("encoding", "utf-8")
+
+    with io.open(filename, encoding=encoding) as f:
+        return f.read()
+
+
+def replace_local_hyperlinks(
+        text, base_url="https://github.com/project-rig/rig/blob/master/"):
+    """Replace local hyperlinks in RST with absolute addresses using the given
+    base URL.
+
+    NOTE: This may need adjusting if further syntax is used.
+    """
+    def get_new_url(url):
+        return base_url + url[2:]
+
+    # Deal with anonymous URLS
+    for match in re.finditer(r"^__ (?P<url>\./.*)", text, re.MULTILINE):
+        orig_url = match.groupdict()["url"]
+        url = get_new_url(orig_url)
+
+        text = re.sub("^__ {}".format(orig_url),
+                      "__ {}".format(url), text, flags=re.MULTILINE)
+
+    # Deal with named URLS
+    for match in re.finditer(r"^\.\. _(?P<identifier>[^:]*): (?P<url>\./.*)",
+                             text, re.MULTILINE):
+        identifier = match.groupdict()["identifier"]
+        orig_url = match.groupdict()["url"]
+        url = get_new_url(orig_url)
+
+        text = re.sub(
+            "^\.\. _{}: {}".format(identifier, orig_url),
+            ".. {}: {}".format(identifier, url),
+            text, flags=re.MULTILINE)
+
+    # Deal with image URLS
+    for match in re.finditer(r"^\.\. image:: (?P<url>\./.*)",
+                             text, re.MULTILINE):
+        orig_url = match.groupdict()["url"]
+        url = get_new_url(orig_url)
+
+        text = text.replace(".. image:: {}".format(orig_url),
+                            ".. image:: {}".format(url))
+
+    return text
+
+long_description = replace_local_hyperlinks(read_file("README.rst"))
```

Comment:
Is it not more conventional to place this line within the setup call? (/nitpick to the XTREAM)